Allow the helm charts to auto generate an edge ingress cert (#919)
greymatter-io/helm-charts: README.md

make credentials
```

### Auto Generate Certificates

> By default, Grey Matter leverages mTLS communications for all traffic, including inbound traffic to the mesh. This means that a user is required to load a certificate into the browser to access Grey Matter. The Grey Matter helm charts have the ability to generate random Ingress certificates and User certificates to ensure unique certs everytime a cluster is launched.

Reviewer: ```suggestion
By default, Grey Matter leverages mutual TLS (mTLS) communications for all traffic, including inbound traffic to the mesh. This means that all `https` requests must include TLS certificates whether that be via a web browser or RESTful client. The Grey Matter helm charts have the ability to generate random Ingress and User certificates to ensure unique certificates every time a cluster is launched. For web based authentication, these certificates can then be imported into a web browser, to access the mesh.
```

I thought this could use a bit of improvement.

> If you want to use the auto generated certs, you can set `.Values.global.auto_generate_edge_certs` to `true` and it will create a self-signed certificate for ingress and one for a user certificate.

Reviewer: ```suggestion
If you want to use the auto generated certs, you can set `.Values.global.auto_generate_edge_certs` to `true` and it will create a self-signed certificate for Ingress and one for a User certificate.
```

Consistent casing with the previous paragraph.

>If you want to provide your own valid certificates for ingress, set `.Values.global.auto_generate_edge_certs` to `false` and provide the cert information in the secrets chart, at `.Values.edge.certificate.ingress`

> To get the user cert, run these commands:

Reviewer: ```suggestion
To get the user certificate, run these commands:
```

Consistent long-hand.

```console
kubectl get secret greymatter-user-cert -o jsonpath="{.data['tls\.crt']}" > tls.crt
kubectl get secret greymatter-user-cert -o jsonpath="{.data['tls\.key']}" > tls.key

> kubectl get secret greymatter-user-cert -o jsonpath="{.data['ca\.crt']}" > ca.crt

Reviewer 2: these don't need base64 decoding?

Author: thanks for the catch.  I haven't actually run these yet.

```

> Then create a new P12 to load into your browser

Reviewer: ```suggestion
Then create a new p12 certificate to load into your browser:
```
